npm/src/clientfileprocessing/index.ts

```diff
@@ -65,14 +73,18 @@ export class ClientFileProcessing {
       )
       const metadata: IFileMetadata[] = await this.clientFileExtractMetadata.extract(
         files,
-        //Temporary function until s3 upload in place
-        this.progressCallback
+        this.progressBarSwitcher
       )
       const tdrFiles = await this.clientFileMetadataUpload.saveClientFileMetadata(
         fileIds,
         metadata
       )
-      this.s3Upload.uploadToS3(consignmentId, tdrFiles, callback, stage)
+      this.s3Upload.uploadToS3(
+        consignmentId,
+        tdrFiles,
+        () => this.updateProgressBarPercentage(100), // Update to 100 as we have finished at this point.
```

Comment:
I don't think this will work as this just sets the display to 100% without updating the progress of the s3 upload. 

See above comments for refactoring the functions slightly, and then call as follows. It's a tricky problem here as we have got two stages to deal with, both independently supply a percent processed, but starting a 0% each time.
```
...
try {
      const fileIds: string[] = await this.clientFileMetadataUpload.saveFileInformation(
        ...
      )

      //Set maximum progress percent to 50 as extract metadata first part of process
      this.setProgressBarDisplay(50)

      const metadata: IFileMetadata[] = await this.clientFileExtractMetadata.extract(
        files,
        this.updateProgressBar
      )
      const tdrFiles = await this.clientFileMetadataUpload.saveClientFileMetadata(
        ...
      )

      //Set maximum progress percent to 100 as upload final part of process
      this.setProgressBarDisplay(100)

      this.s3Upload.uploadToS3(
        consignmentId,
        tdrFiles,
        this.updateProgressBar,
        stage
      )
...
```